caluma/workflow/filters.py

```diff
@@ -53,6 +56,48 @@ class CaseFilterSet(MetaFilterSet):
 
     has_answer = HasAnswerFilter(document_id="document__pk")
     status = case_status_filter(lookup_expr="in")
+    order_by_question_answer_value = CharFilter(
+        method="filter_order_by_question_answer_value",
+        label=(
+            "Expects a question slug. If the slug is prefixed with a dash, "
+            "the order will be reversed\n\n"
+            "For file questions, the filename is used for sorting.\n\n"
+            "Table questions are not supported at this time."
+        ),
+    )
+
+    @staticmethod
+    def filter_order_by_question_answer_value(queryset, _, value):
+        order_by = "-order_value" if value.startswith("-") else "order_value"
+        value = value.lstrip("-")
+
+        path = value.split(".")
+        question_slug = path.pop(-1)
+
+        # Based on question type, set answer field to use for sorting
+        not_supported = (Question.TYPE_TABLE,)
+        question = Question.objects.get(slug=question_slug)
+        answer_value = "value"
+        if question.type in not_supported:
+            raise RuntimeError(
+                f'Questions with type "{question.type}" are not supported '
+                f'by "filterOrderByQuestionAnswerValue"'
+            )
+        elif question.type == Question.TYPE_DATE:
+            answer_value = "date"
+        elif question.type == Question.TYPE_FILE:
+            answer_value = "file__name"
```

Comment:
This question type to value field mapping is already at several places in the codebase, right? It might be worth extracting this into a function?

Reply:
You are right, but it is much simpler to address this, after #530 is resolved.

Reply:
:+1: Alright, lets solve it in another PR then.